### PR TITLE
feat: add golang functionality

### PR DESCRIPTION
**Pull request by mblume97**

### CODE REVIEW

File: generators/language-go/index.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
+/*
+ * © Copyright IBM Corp. 2017, 2018
```
Comment (jkingoliver) [inline generators/language-go/index.js]: for a brand-new file, you only need to use `2018`. Please delete the `2017`

Comment (mblume97) [reply generators/language-go/index.js]: done!

File: test/app-golang-test.js [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+		helpers
+			.run(path.join(__dirname, GENERATOR_PATH))
+			.inTmpDir()
+			.withOptions({
+				bluemix: JSON.stringify(optionsBluemix)
+			})
+			.then((tmpDir) => {
+				console.info(tmpDir);
+
+				// Gopkg.toml doesn't have any SDKs
```
Comment (jkingoliver) [inline test/app-golang-test.js]: delete this commented code

File: test/app-golang-test.js [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+	it('Can run successful generation and create files', () => {
+		yassert.file('server');
+		yassert.file('server/config');
+		yassert.file(SERVER_MAPPINGS_JSON);
+		yassert.file(SERVER_LOCALDEV_CONFIG_JSON);
+		yassert.file('services/services.go');
+	});
+
+	// it('Can add Apache Spark instrumentation', () => {
```
Comment (jkingoliver) [inline test/app-golang-test.js]: delete all these commented out sections for services go doesn't support

File: test/app-golang-test.js [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+	// it('Can add Finance - Simulated Instrument Analytics instrumentation', () => {
+	// 	testAll('finance-simulated-instrument-analytics', {
+	// 		finance_simulated_instrument_analytics_uri: optionsBluemix.simulatedInstrumentAnalytics.uri,
+	// 		finance_simulated_instrument_analytics_accessToken: optionsBluemix.simulatedInstrumentAnalytics.accessToken
+	// 	});
+	// });
+
+	it('Can add Watson - Assistant instrumentation', () => {
+		testAll('watson-assistant', {
+			watson_assistant_url: optionsBluemix.conversation.url,
```
[the remark this answers is not in the document]
Comment (mblume97) [reply test/app-golang-test.js]: yes the string is definitely conversation
https://github.com/ibm-developer/generator-ibm-service-enablement/blob/develop/test/app-swift-kitura-test.js#L116

**Pull request merged**